src/components/map/layers/NeighborLinkLayer.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { Component } from 'react';
+import { Polyline } from 'react-leaflet';
+import { inject, observer } from 'mobx-react';
+import IRoutePathLink from '~/models/IRoutePathLink';
+import INode from '~/models/INode';
+import { RoutePathStore, NeighborToAddType } from '~/stores/routePathStore';
+import { MapStore } from '~/stores/mapStore';
+import RoutePathLinkService from '~/services/routePathLinkService';
+import INeighborLink from '~/models/INeighborLink';
+import NodeMarker from './mapIcons/NodeMarker';
+
+const USED_NEIGHBOR_COLOR = '#0dce0a';
+const UNUSED_NEIGHBOR_COLOR = '#fc383a';
+
+interface IRoutePathLayerProps {
+    routePathStore?: RoutePathStore;
+    mapStore?: MapStore;
+}
+
+@inject('routePathStore', 'mapStore')
+@observer
+class NeighborLinkLayer extends Component<IRoutePathLayerProps> {
```

Comment:
I would name this as "RoutePathNeighborLinkLayer" to indicate that it really is related to routePathLayer

Reply:
I also would put this to layers/edit/routePathNeighborLinkLayer